src/App.tsx

```diff
@@ -14,37 +16,105 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const REVERSE = false;
+
+enum SortType {
+  RESET_VALUE = '',
+  SORT_BY_NAME = 'name',
+  SORT_BY_LENGTH = 'length',
+}
+
+interface SortCriteria {
+  sortBy: SortType;
+  isReversed: boolean;
+}
+
+function sortGoods(
+  goods: string[],
+  { sortBy: criteria, isReversed }: SortCriteria,
+) {
+  let sortedGoods;
+
+  switch (criteria) {
+    case SortType.SORT_BY_NAME:
+      sortedGoods = goods.toSorted((a, b) => a.localeCompare(b));
```

Comment:
It seems there's a typo in the method name `toSorted`. In JavaScript arrays, the correct method name is `sort`. Also, direct DOM manipulation is not recommended in React. Use state and effects to handle rendering logic.